src/Grid/GridHeader.js

```diff
@@ -63,71 +62,87 @@ class GridHeader extends React.Component {
+    let secondValue = '';
+
+    switch (this.state.columnType){
+    case 'datetime':
+    case 'date':
+    case 'datetimeutc':
+      firstValue = moment().format();
+      secondValue = moment().format();
+      break;
+    default:
     }
 
     this.setState({
       [this.state.activeFilter]: 'None',
-      [`${this.state.activeFilter}Value`]: value1,
-      [`${this.state.activeFilter}Value2`]: value2
+      [`${this.state.activeFilter}Value`]: firstValue,
+      [`${this.state.activeFilter}Value2`]: secondValue
     }, () => {
-      this.filterHandler(value1, value2, false);
+      this.filterHandler(firstValue, secondValue, false);
     });
   }
 
   handleApply = () => {
-    let value1 = this.state[`${this.state.activeFilter}Value`];
-    let value2 = value2 === null ? null : this.state[`${this.state.activeFilter}Value2`];
-
-    if(this.state.columnType === 'numeric'){
-      value1 = parseFloat(value1);
-      value2 = parseFloat(value2);
-    }
-    else if(this.state.columnType === 'boolean'){
-      value1 = (value1 === 'true');
-      value2 = 1;
+    let firstValue = this.state[`${this.state.activeFilter}Value`];
+    let secondValue = secondValue !== null && this.state[`${this.state.activeFilter}Value2`];
+
+    switch (this.state.columnType){
+    case 'numeric':
+      firstValue = parseFloat(firstValue);
+      secondValue = parseFloat(secondValue);
+      break;
+    case 'boolean':
+      firstValue = (firstValue === 'true');
+      secondValue = 1;
+      break;
+    default:
     }
     
-    this.filterHandler(value1, value2, true);
+    this.filterHandler(firstValue, secondValue, true);
   }
 
-  filterHandler = (value1, value2, hasFilter) => {
-    const dataSource = Object.assign([{}], this.state.dataSource);
-
-    dataSource.columns.forEach( (row, i) => {
+  filterHandler = (firstValue, secondValue, hasFilter) => {
+    this.state.dataSource.columns.forEach( (row, i) => {
       if(row.Name === this.state.activeFilter){
-        dataSource.columns[i].Filter.Text = value1;
-        dataSource.columns[i].Filter.Operator = this.state[this.state.activeFilter];
-        dataSource.columns[i].Filter.HasFilter = hasFilter;
-        if(value2 !== undefined){
-          dataSource.columns[i].Filter.Argument = [value2];
+        row.Filter.Text = firstValue;
+        row.Filter.Operator = this.state[this.state.activeFilter];
+        row.Filter.HasFilter = hasFilter;
+        if(secondValue !== undefined){
+          row.Filter.Argument = [secondValue];
         }
       }
     });
 
     this.state.dataSource.filter(this.state.rowsPerPage, this.state.page);
   }
 
-  sortHandler = property => {
-    const dataSource = Object.assign([{}], this.state.dataSource);
-
-    dataSource.columns.forEach( (row, i) => {
-      if(row.SortOrder === 1){
-        dataSource.columns[i].SortOrder = -1;
-      }
+  sortHandler = (event, property) => {
+    let sortOrder = 1;
 
+    this.state.dataSource.columns.forEach( row => {
```

Comment:
Please rename that "row" variable, you're doing forEach over columns array, not rows.